Assets/Scripts/UI/Common/InventoryPanel.cs

```diff
@@ -20,10 +27,15 @@ public static InventoryPanel create(in Inventory inventory, RectTransform parent
             return inventoryPanel;
         }
 
+        /// <summary>
+        /// Initializes data of the inventory panel
+        /// </summary>
         private void setData(Inventory inventory, RectTransform parent)
         {
             this.tabName = "Inventory";
             this.inventory = inventory;
+            inventory = new Inventory();
```

Comment:
I'm a bit confused at whats going on here, could you explain please?

Reply:
Generally speaking I have tried to **seperate UI components** from **data components**. . This shows an example of how the `InventoryPanel` (UI class), instantiating the underlying inventory data model.

This convention is mainly to seperate out classes that use unity objects and those which represent the object models.

Reply:
Admitidly, I just realised I pass in an inventory, but then  instantiate a new one.  I think I had intention to have an invetory be passed in from an external  source, but then just hacked together an inline  instantiation.  I will make this clearer